### PR TITLE
Reformatting from examples_utils to graphcore_cloud_tools

### PR DESCRIPTION
**Pull request by hiteshk-gc**
Since this was a copy of examples_utils, moving everything from this repo to the name/package name/concept of graphcore_cloud_tools

### CODE REVIEW

**Comment by payoto** [comment]
Can you run the tests locally? I've asked for actions but we don't have them yet

File: .github/workflows/copyright-header-check.yaml [hunk 1]
```diff
@@ -10,4 +10,4 @@ jobs:
       - name: Checkout repo
         uses: actions/checkout@v1
       - name: Run test_copyright.py
-        run: python3 examples_utils/testing/test_copyright.py
+        run: python3 graphcore_cloud_tools/testing/test_copyright.py
```
Comment (payoto) [inline .github/workflows/copyright-header-check.yaml:13]: these can stay on examples-utils as we're going to strip it out I would think

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -26,8 +26,8 @@ repos:
     - id: copyright-header-check
       name: Copyright header check
       description: Ensures that files have the proper copyright line at the top
-      entry: python3 -m examples_utils test_copyright --amend
+      entry: python3 -m graphcore_cloud_tools test_copyright --amend
       pass_filenames: false
       language: python
       additional_dependencies:
-        - 'git+https://github.com/graphcore/examples-utils@latest_stable'
+        - 'git+https://github.com/graphcore/graphcore-cloud-tools@latest_stable'
```
Comment (payoto) [inline .pre-commit-config.yaml:33]: Same can stay on examples-utils

**Review by payoto — COMMENTED** [review]
LGTM pending a test run to make sure that you've caught as many as you could

File: .github/workflows/copyright-header-check.yaml (hunk 1 above)
Comment (hiteshk-gc) [reply .github/workflows/copyright-header-check.yaml:13]: True, I did global search/replace but forgot about these

**Comment by payoto** [comment]
What was deleted to justify the reduction in lines of code?

Screenshot 2023-07-03 at 17 30 56

**Comment by hiteshk-gc** [comment]
I deleted the benchmarking and platform assessment tools too, I think examples_utils and gc cloud tools can be entirely seperate, and parts where the platform assessment are needed should import examples_utils as well. 

This keeps user notebooks as light as possible while our infra can import one more thing without any more potential clashes than there already are

File: .github/workflows/copyright-header-check.yaml
Comment (payoto) [inline .github/workflows/copyright-header-check.yaml:1]: Can you leave the CI definition? We still need those (applies for all files in .github/workflows)

File: .pre-commit-config.yaml
Comment (payoto) [inline .pre-commit-config.yaml:1]: We can still use pre-commit as well

**Review by payoto — CHANGES_REQUESTED** [review]
I agree about stripping things out to keep the dependency lightweight, but this is very hard to review as the combined move + deletion makes it hard to identify what has been removed or not. To approve the refactor I need to see that the notebook logger and the paperspace utils still work in a  test that makes that obvious.

In terms of things to keep remove I think: 
- should stay:
  - github workflows
  - pre-commit config
- should go:
  - pinned requirements
  - testing
  - parsing

**Comment by hiteshk-gc** [comment]
I understand, I've tested locally and proven:

logger installs via pip (had to use branch since this PR hasnt landed):
gct_install

Logger runs locally in debug mode (prints out payload):
gct_logger_debug_mode

Logger also uploads as intended (with some tweaks to make it run outside of paperspace):
gct_logger_works

**Comment by hiteshk-gc** [comment]
I will try to do the same for paperspace utils but I dont have experience with those, so will take a short while

**Comment by payoto** [comment]
In terms of testing with @marcins-gc's PR you could move one of his PRs to run from this repo and this branch. This will then trigger the probe in the PS env

File: graphcore_cloud_tools/testing/__init__.py
Comment (payoto) [inline graphcore_cloud_tools/testing/__init__.py:1]: I think the testing folder can be imported from examples-utils we don't need it here, it's just a dev dependency?

**Review by payoto — APPROVED** [review]
Thanks

**Comment by hiteshk-gc** [comment]
Cant get pre-commit to run, I suspect thats because the repo master branch is missing the file since its been renamed/re-pathed in this branch and hence it cant find the location in master?

File: graphcore_cloud_tools/testing/__init__.py
Comment (hiteshk-gc) [reply graphcore_cloud_tools/testing/__init__.py:1]: I've removed test_command but kept test-copyright, or should we import that too?

File: .pre-commit-hooks.yaml [hunk 3]
```diff
@@ -2,5 +2,10 @@
...
-  entry: examples_utils/precommit/pinned_requirements/entrypoint.sh
-  files: 'requirements*.txt'
\ No newline at end of file
+  entry: graphcore_cloud_tools/precommit/pinned_requirements/entrypoint.sh
+  files: 'requirements*.txt'
+- id: copyright-header-check
+  name: Copyright header check
+  description: Ensures that files have the proper copyright line at the top
+  language: python
+  entry: python3 -m graphcore_cloud_tools test_copyright
```
Comment (payoto) [inline .pre-commit-hooks.yaml:11]:
```suggestion
  entry: python3 -m examples_utils test_copyright
  additional_dependencies:
    - 'git+https://github.com/graphcore/examples-utils@latest_stable'
```

File: .github/workflows/copyright-header-check.yaml
Comment (payoto) [inline .github/workflows/copyright-header-check.yaml:1]: I think you can remove this if you use the pre-commit version?

I think you need to install examples utils and use that to do the copyright check?

Comment (hiteshk-gc) [reply .github/workflows/copyright-header-check.yaml:1]: Thanks, just added your changes, if the step passes I'll remove the file from here and merge

Comment (hiteshk-gc) [reply .github/workflows/copyright-header-check.yaml:1]: Oh I just realised what you meant! 😂

**Pull request merged**